src/routes/safe/components/Apps/index.tsx

```diff
@@ -38,26 +42,62 @@ const CenteredMT = styled.div`
   margin-top: 5px;
 `
 
-type AppsProps = {
-  closeModal: () => void
-  openModal: (modal: OpenModalArgs) => void
+type ConfirmTransactionModalState = {
+  isOpen: boolean
+  txs: Transaction[]
+  requestId: RequestId | undefined
 }
 
-const Apps = ({ closeModal, openModal }: AppsProps): React.ReactElement => {
+const INITIAL_CONFIRM_TX_MODAL_STATE: ConfirmTransactionModalState = {
+  isOpen: false,
+  txs: [],
+  requestId: undefined,
+}
+
+const Apps = (): React.ReactElement => {
   const { appList, loadingAppList, onAppToggle, onAppAdded } = useAppList()
 
   const [appIsLoading, setAppIsLoading] = useState<boolean>(true)
   const [selectedAppId, setSelectedAppId] = useState<string>()
+  const [confirmTransactionModal, setConfirmTransactionModal] = useState<ConfirmTransactionModalState>(
+    INITIAL_CONFIRM_TX_MODAL_STATE,
+  )
   const iframeRef = useRef<HTMLIFrameElement>()
 
   const granted = useSelector(grantedSelector)
   const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const safeName = useSelector(safeNameSelector)
   const network = useSelector(networkSelector)
   const ethBalance = useSelector(safeEthBalanceSelector)
 
+  const openConfirmationModal = useCallback(
+    (txs: Transaction[], requestId: RequestId) =>
+      setConfirmTransactionModal({
+        isOpen: true,
+        txs,
+        requestId,
+      }),
+    [setConfirmTransactionModal],
+  )
+  const closeConfirmationModal = useCallback(() => setConfirmTransactionModal(INITIAL_CONFIRM_TX_MODAL_STATE), [
```

Comment:
Should we notify the SafeApp that the confirmation process was canceled (we talked about that with the GP team)

Reply:
Yes, and I will, but the PR is already very big and I wanted to do it in a separate one. AFAIK knowing if the user pressed confirm button is top 1 blocker for the GP team

Reply:
could we create a follow up issue for that?

Reply:
https://github.com/gnosis/safe-apps-sdk/issues/37